src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -477,6 +484,55 @@ public AgentSettings LoadSettings()
             }
         }
 
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            bool enableAutoLogon = command.GetEnableAutoLogon();
+            if(!enableAutoLogon)
+            {
+                return;
+            }
+
+            try
+            {
+                AssertAdminAccess(command);
+                
+                var iConfigManager = HostContext.GetService<IInteractiveSessionConfigurationManager>();
+                iConfigManager.Configure(command);
+
+                var processHelper = HostContext.GetService<IProcessInvoker>();
+                if(iConfigManager.RestartNeeded())
+                {
+                    _term.WriteLine(StringUtil.Loc("RestartMessage"));
+                    var shallRestart = command.GetRestartPermission();
+                    if(shallRestart)
+                    {
+                        processHelper.ExecuteAsync(
+                            workingDirectory: string.Empty,
+                            fileName: "shutdown.exe",
+                            arguments: "-r -t 0",
+                            environment: null,
+                            cancellationToken: CancellationToken.None).Wait();                        
+                    }
+                }
+                else
+                {
+                    var startupProcessPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "agentservice.exe");
+                    processHelper.ExecuteAsync(
+                        workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Bin),
+                        fileName: startupProcessPath,
+                        arguments: "runasprocess",
+                        environment: null,
+                        cancellationToken: CancellationToken.None,
+                        waitForExit: false).Wait();
+                }
+            }
+            catch(Exception ex)
+            {
+                Trace.Error(ex);
+                _term.WriteLine(StringUtil.Loc("AutoLogonConfigurationFailureMessage"));
```

Comment:
AutoLogon config fails. Agent is configured and user can reattempt to configure autologon on continue with run.cmd to have an agent with interactivesession capability.